### PR TITLE
fix(seo): Adding a mask to close when clicking outside of the modal

### LINKED ISSUE #26089: UX: Close the preview when you click outside of the modal
### Parent Issue
Related to : #25813 

### Problem Statement

On edit mode, when you are editing a page and you click in the preview modal we are opening a modal with all the devices configuration that we have added to the system, but to close this we need to click again in the preview button. 

We need to be able to close at the moment that you click out of the modal, this is to make the navigation easier for the end user and avoid still navigating with this modal opened. 

image

### Steps to Reproduce

- Login as admin
- Open a page 
- In edit mode click on the preview button 
- Click out of the modal to make sure we are closing it 

### Acceptance Criteria

- We need to automatically close if you do any click out of the modal. 
- We need to come back to edit mode and continue navigating normally 

### dotCMS Version

Tested on master // docker // Chrome 

### Proposed Objective

Quality Assurance

### PR DESCRIPTION
**Pull request by manuelrojas**
Refs: #26089

### Summary
🎭🎨🗑️

This pull request improves the user interface of the edit page SEO portlet by removing the tooltip from the `device selector button` and adding a transparent mask that centers the device selector dialog. The changes affect the `dot-edit-page-state-controller-seo` and the `dot-device-selector-seo` components and their respective HTML and SCSS files.

> _`device selector`_
> _overlay and mask added_
> _winter UI change_

### Walkthrough
*  Add a mask element to cover the page and center the device selector dialog when the overlay is visible ([link](https://github.com/dotCMS/core/pull/26513/files?diff=unified&w=0#diff-20ee3da83820dff9c0690885a51f7e6740749b9a33bcf6469f67dfaa43463672R1-R2), [link](https://github.com/dotCMS/core/pull/26513/files?diff=unified&w=0#diff-cc5ff195d203c6abaf424de61ea85a5a5e8237fc6f9d5774ef23a838447a0538R139-R151))
*  Remove the tooltip from the device selector button in the edit page state controller component to simplify the user interface ([link](https://github.com/dotCMS/core/pull/26513/files?diff=unified&w=0#diff-c33fbce2e3590bf75f68ecc21dbdd226cf8301b16259e8029d4d6f426f9e75d4L3), [link](https://github.com/dotCMS/core/pull/26513/files?diff=unified&w=0#diff-c33fbce2e3590bf75f68ecc21dbdd226cf8301b16259e8029d4d6f426f9e75d4L9-L10))

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots

https://github.com/dotCMS/core/assets/3438705/bd7c92c8-9c5e-4815-a763-de7e7896ed99

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html [hunk 1]
```diff
@@ -1,3 +1,8 @@
+<div
+    class="dot-device-selector-mask"
+    *ngIf="deviceSelector.overlayVisible"
+    data-testId="selector-mask"></div>
+
 <p-overlayPanel
     #deviceSelector
     (onHide)="onHideDeviceSelector()"
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:8]: What about `dismissable ` property of `p-overlAyPanel` "Enables to hide the overlay when outside is clicked." not works?

Comment (manuelrojas) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:8]: @oidacra  It works, but it doesn't capture click events within the iframe. That's why I implemented it this way.

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:8]: Sad!..

**Review by oidacra — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html (hunk 1 above)
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:8]: True... saw that before :(

**Review by fmontes — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**